# Masters-Project
Queen Mary University of London

Bioinformatics Research Project</td></tr>
Module code: BIO702P

Machine Learning to Predict Causal and Druggable Cardiovascular Disease Variants Following Genome-wide Association Study

Master of Science (MSc) Bioinformatics

August 2025


Code Availability</td></tr>
All code used in the project has been deposited in GitHub and can be found in at https://github.com/Radragon5/Masters-Project

Table 2 - Summary of Programs Used in this Project

Abbreviated</td><td>Programming</td><td>Full</td></tr>
Name</td><td></td><td>Language</td><td>Name</td></tr>

<table>
  <tr><td>
    <th>Abbreviated Name</th>
    <th>Programming Language</th>
    <th>Full Name</th>
  </td></tr>
  <tr><td>
    <td>Alfreds Futterkiste</td>
    <td>Maria Anders</td>
    <td>Germany</td>
  </td></tr>
  <tr><td>
    <td>Centro comercial Moctezuma</td>
    <td>Francisco Chang</td>
    <td>Mexico</td>
  </td></tr>
<tr><td>Step 1-1</td></td><td><td>R</td></td><td><td>Step 1-1 Find Lead SNPs</td></tr>
<tr><td>Step 1-2</td><td>R</td><td>Step 1-2 Find SNPs Within 500kb</td></tr>
<tr><td>Step 2-1</td><td>R</td><td>Step 2-1 Get ClinVar Phenotype List</td></tr>
<tr><td>Step 3-1</td><td>R</td><td>Step 3-1 Get RSIDs From Biomart</td></tr>
<tr><td>Step 3-2</td><td>R</td><td>Step 3-2 Get RSIDs from BIM files</td></tr></td></tr>
<tr><td>Step 3-3</td><td>R</td><td>Step 3-3 Split VCF files into smaller chunks</td></tr>
<tr><td>Step 3-4</td><td>R</td><td>Step 3-4 Get RSIDs From VCF file</td></tr>
<tr><td>Step 3-5</td><td>R</td><td>Step 3-5 Get PharmGKB Phenotype List</td></tr>
<tr><td>Step 4-1</td><td>R</td><td>Step 4-1 Get Causal and Druggable Mappings</td></tr>
<tr><td>Step 5-1</td><td>R</td><td>Step 5-1 VEP Model Parameters before Cleaning</td></tr>
<tr><td>Step 5-2</td><td>R</td><td>Step 5-2 Create VCF Files for DeepSEA</td></tr>
<tr><td>Step 5-3</td><td>R</td><td>Step 5-3 Recombine Results Files</td></tr>
<tr><td>Step 5-4</td><td>R</td><td>Step 5-4 Collapse VEP Results to single lines</td></tr>
<tr><td>Step 5-5</td><td>R</td><td>Step 5-5 VEP Model Parameters before Cleaning</td></tr>
<tr><td>Step 5-6</td><td>R</td><td>Step 5-6 Create REVEL relationships file</td></tr>
<tr><td>Step 5-7</td><td>R</td><td>Step 5-7 Create VCF File Chunks CADD</td></tr>
<tr><td>Step 5-8</td><td>R</td><td>Step 5-8 Recombine VCF File Chunks</td></tr>
<tr><td>Step 5-9</td><td>R</td><td>Step 5-9 Reformat CADD Results</td></tr>
<tr><td>Step 5-10</td><td>R</td><td>Step 5-10 Combine resources</td></tr>
<tr><td>Step 6-1</td><td>R</td><td>Step 6-1 Pairwise Analysis</td></tr>
<tr><td>Step 6-2</td><td>R</td><td>Step 6-2 Missingness Analysis</td></tr>
<tr><td>Step 6-3</td><td>Python</td><td>Step 6-3 Impute KNN</td></tr>
<tr><td>Step 7-1</td><td>R</td><td>Step 7-1 Skewness and Kurtosis</td></tr>
<tr><td>Step 7-2</td><td>Python</td><td>Step 7-2 PCA</td></tr>
<tr><td>Step 8</td><td>Python</td><td>Step 8 ML Model</td></tr>
<tr><td>Step 9-1</td><td>R</td><td>Step 9-1 Annotate SNPs to Genes</td></tr>
<tr><td>Step 9-2</td><td>R</td><td>Step 9-2 Analyse variant scores</td></tr>
<tr><td>Step 9-3</td><td>Python</td><td>Step 9-3 Compare Model to GWAS</td></tr>

</table>

All programs used in this project listed above are available in GitHub at the indicated URL. Each program is prefixed by Step x-y, indicating their logical grouping (x) and order of usage (y).

