# Masters-Project
Queen Mary University of London

Bioinformatics Research Project</tr>
Module code: BIO702P

Machine Learning to Predict Causal and Druggable Cardiovascular Disease Variants Following Genome-wide Association Study

Master of Science (MSc) Bioinformatics

August 2025


Code Availability</tr>
All code used in the project has been deposited in GitHub and can be found in at https://github.com/Radragon5/Masters-Project

Table 2 - Summary of Programs Used in this Project

Abbreviated	Programming	Full</tr>
Name		Language	Name</tr>

<table>
  <tr>
    <th>Abbreviated Name</th>
    <th>Programming Language</th>
    <th>Full Name</th>
  </tr>
  <tr>
    <td>Alfreds Futterkiste</td>
    <td>Maria Anders</td>
    <td>Germany</td>
  </tr>
  <tr>
    <td>Centro comercial Moctezuma</td>
    <td>Francisco Chang</td>
    <td>Mexico</td>
  </tr>



<tr>Step 1-1	R			Step 1-1 Find Lead SNPs</tr>
<tr>Step 1-2	R			Step 1-2 Find SNPs Within 500kb</tr>
<tr>Step 2-1	R			Step 2-1 Get ClinVar Phenotype List</tr>
<tr>Step 3-1	R			Step 3-1 Get RSIDs From Biomart</tr>
<tr>Step 3-2	R			Step 3-2 Get RSIDs from BIM files</tr></tr>
<tr>Step 3-3	R			Step 3-3 Split VCF files into smaller chunks</tr>
<tr>Step 3-4	R			Step 3-4 Get RSIDs From VCF file</tr>
<tr>Step 3-5	R			Step 3-5 Get PharmGKB Phenotype List</tr>
<tr>Step 4-1	R			Step 4-1 Get Causal and Druggable Mappings</tr>
<tr>Step 5-1	R			Step 5-1 VEP Model Parameters before Cleaning</tr>
<tr>Step 5-2	R			Step 5-2 Create VCF Files for DeepSEA</tr>
<tr>Step 5-3	R			Step 5-3 Recombine Results Files</tr>
<tr>Step 5-4	R			Step 5-4 Collapse VEP Results to single lines</tr>
<tr>Step 5-5	R			Step 5-5 VEP Model Parameters before Cleaning</tr>
<tr>Step 5-6	R			Step 5-6 Create REVEL relationships file</tr>
<tr>Step 5-7	R			Step 5-7 Create VCF File Chunks CADD</tr>
<tr>Step 5-8	R			Step 5-8 Recombine VCF File Chunks</tr>
<tr>Step 5-9	R			Step 5-9 Reformat CADD Results</tr>
<tr>Step 5-10	R			Step 5-10 Combine resources</tr>
<tr>Step 6-1	R			Step 6-1 Pairwise Analysis</tr>
<tr>Step 6-2	R			Step 6-2 Missingness Analysis</tr>
<tr>Step 6-3	Python		Step 6-3 Impute KNN</tr>
<tr>Step 7-1	R			Step 7-1 Skewness and Kurtosis</tr>
<tr>Step 7-2	Python		Step 7-2 PCA</tr>
<tr>Step 8		Python		Step 8 ML Model</tr>
<tr>Step 9-1	R			Step 9-1 Annotate SNPs to Genes</tr>
<tr>Step 9-2	R			Step 9-2 Analyse variant scores</tr>
<tr>Step 9-3	Python		Step 9-3 Compare Model to GWAS</tr>

</table>

All programs used in this project listed above are available in GitHub at the indicated URL. Each program is prefixed by Step x-y, indicating their logical grouping (x) and order of usage (y).

